src/main/java/org/sonatype/goodies/packageurl/PackageUrlParser.java

```diff
@@ -155,17 +149,21 @@ static String parseVersion(@Nullable final String value) {
       return null;
     }
 
-    Map<String, String> pairs = QUALIFIER_SPLITTER.split(value);
-    Map<String, String> result = new LinkedHashMap<>(pairs.size());
-    for (Map.Entry<String, String> entry : pairs.entrySet()) {
-      String v = entry.getValue();
-
-      // qualifiers with empty values should be skipped
+    String[] pairs = value.split("&");
+    Map<String, String> result = new LinkedHashMap<>(pairs.length);
+    for (String pair : pairs) {
+      String[] split = pair.split("=", 2);
+      if (split.length == 1) {
```

Comment:
this should probably be `if (split.legnth != 2)`

Reply:
This will change really soon with the "as is" mode, so in the meanwhile I have added a comment
`// Splits the pair into either one or two pieces.`

Reply:
@andrewstein I think the asis stuff we talked about was only in reference to the transformation, I think here the current/desired behavior is still to obey the spec when parsing and discard qualifiers with blank values.